File: solutions/elixir/roman-numerals/1/lib/roman_numerals.ex
defmodule RomanNumerals do
  @doc """
  Convert the number to a roman number.
  """
  @numerals ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
  @values [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]

  @spec numeral(pos_integer) :: String.t()
  def numeral(number) do
    numeral(@numerals, @values, number, "")
  end

  def numeral([], [], number, str) do
    str
  end

  def numeral([first_num | tail_num], [first_val | tail_val], number, str) do
    if first_val > number do
      numeral(tail_num, tail_val, number, str)
    else
      numeral([first_num | tail_num], [first_val | tail_val], number - first_val, str <> first_num)
    end
  end
end

File: solutions/python/roman-numerals/2/roman_numerals.py
ROMAN_NUMERALS = {
    1000: "M",
    900: "CM",
    500: "D",
    400: "CD",
    100: "C",
    90: "XC",
    50: "L",
    40: "XL",
    10: "X",
    9: "IX",
    5: "V",
    4: "IV",
    1: "I",
}


def roman(number):
    return "".join(
        roman * count
        for val, roman in ROMAN_NUMERALS.items()
        if (number := number - val * (count := number // val)) or True
    )
